include/mp/proxy-types.h

```diff
@@ -955,22 +955,26 @@ void CustomBuildField(TypeList<LocalType> local_type,
 template <typename Accessor, typename LocalType, typename ServerContext, typename Fn, typename... Args>
 void PassField(TypeList<LocalType*>, ServerContext& server_context, const Fn& fn, Args&&... args)
 {
-    InvokeContext& invoke_context = server_context;
-    boost::optional<Decay<LocalType>> param;
     const auto& params = server_context.call_context.getParams();
     const auto& input = Make<StructField, Accessor>(params);
-    bool want = input.want();
-    if (want) {
-        MaybeReadField(std::integral_constant<bool, Accessor::in>(), TypeList<LocalType>(), invoke_context, input,
-            Emplace<decltype(param)>(param));
-        if (!param) param.emplace();
+
+    if (!input.want()) {
+        fn.invoke(server_context, std::forward<Args>(args)..., nullptr);
+        server_context.call_context.getResults();
```

Comment:
Should be safe to drop this line. Can be done in a followup though since keeping it makes the refactoring more obvious.

Reply:
https://github.com/chaincodelabs/libmultiprocess/pull/18

Cheerz!